core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -566,6 +574,13 @@ private[spark] class ExecutorAllocationManager(
     if (executorIds.contains(executorId)) {
       executorIds.remove(executorId)
       removeTimes.remove(executorId)
+      if (!removedExecutorIds.offer(executorId)) {
```

Comment:
I get it. This `removedExecutorIds`  used for cache status of removed executor id.
If  `SparkListenerExecutorRemoved` event is posted before `SparkListenerTaskStart` event, `executorIds` will remove  executor id.Then the method `onTaskStart` will adding the removed executor id back to `executorIds`.

Reply:
Yes,this bug will lead to executorIds contains some removed executor id and newExecutorTotal is an incorrect number.

Reply:
@beliefer Yes, that's it!